src/emit/solana.rs

```diff
@@ -868,6 +869,15 @@ impl SolanaTarget {
 
                 self.storage_free(binary, &field.ty, data, offset, function, zero, ns);
             }
+        } else if matches!(ty, Type::Address(_)) {
```

Comment:
I think you need `Type::Contract(_)` here too

Reply:
Fixed